### ISSUE TITLE
Add mock requests tests

### ISSUE THREAD

**Issue by gtalarico** [gtalarico/airtable-python-wrapper#3]

**Comment by gtalarico** [comment]
I have not had much luck revising tests to use Mock requests, so any help on this is greatly appreciated.

1. The current test suite attempts to reset the airtable state before tests are run, but the state is still fragil.
2. Current test suite relies on an airtable under my account to run tests. Anyone running test suite would need to either get my API key, or rebuild a similar table so they can run the tests. Both scenarios are not feasible. 

I like the idea that some of the tests call the actual API, but the core of the suite could be tested with mock objects. (requests mocker, or similar)

This seems like it could also be a reasonable solution:
https://github.com/kevin1024/vcrpy


https://github.com/kevin1024/vcrpy

**Comment by bpeterso2000** [comment]
Hi @gtalarico, thank you so much for creating this project.  If you're interested I have a full suite of tests with 100% code coverage for this project; they don't require a connection to Airtable so they will be suitable for setting up continuous integration.  I also included _tox_ for local testing against different Python versions.  I will submit a PR so that you can take a look at it and see what you think.

If you're willing to entertain the idea, I'm also interested in helping with some of the other project issues here as well, but wanted to have a full suite of unit tests in place first without a dependency on API keys, Airtable setup, etc.

Thanks,
Brian

**Comment by gtalarico** [comment]
That would be amazing Brian.
Thank you for the work. 
I would be happy to add you as maintainer and give you repo access if you would to help. 

I added some comments to your pr. 
Tldr, I would to get your tests in, but we would need to resolve conflicts with the work I already merged.

**Comment by bpeterso2000** [comment]
Sounds good, I'll take a look your comments and start working on merging in the changes.
    On Wednesday, June 5, 2019, 6:11:41 PM EDT, Gui Talarico <notifications@github.com> wrote:  
 
 
That would be amazing Brian.
Thank you for the work.
I would be happy to add you as maintainer and give you repo access if you would to help.

I added some comments to your pr.
Tldr, I would to get your tests in, but we would need to resolve conflicts with the work I already merged.

**Comment by gtalarico** [comment]
PS: Fixed coverage reporting, should show correctly
https://coveralls.io/builds/23824999/source?filename=airtable/airtable.py

**Comment by bpeterso2000** [comment]
Added test coverage for `Airtable.match`.

**Comment by bpeterso2000** [comment]
In the `todo` file there's a task to deprecate `mirror`; as a result, I didn't write a test for it.   Do you think we should deprecate `mirror` now?  Or do you think it's better to go ahead and write a test for it?

**Comment by gtalarico** [comment]
Haven't thought about it much. I personally haven't used it, but maybe others do.
Are you using it?

**Comment by bpeterso2000** [comment]
No, I'm not using `mirror`, I can open a separate issue to deprecate it and we can give everyone a couple of weeks to comment on it.  It seems kind of dangerous in that it wipes out all the old data in the table.

I think we can close this issue since the mock request tests are done (except `mirror`.)  I have a couple more tests I'll submit, but they're not related to mock request tests, they're related to exception handling and dealing with whether or not it's using `ironpython`.

**Comment by gtalarico** [comment]
@bpeterso2000 thanks for all your contribution and tests. closing this
